test/org/sagebionetworks/bridge/upload/IosSchemaValidationHandler2Test.java

```diff
@@ -254,11 +274,56 @@ public void survey() throws Exception {
                 "}";
         JsonNode bazAnswerJsonNode = BridgeObjectMapper.get().readTree(bazAnswerJsonText);
 
-        context.setJsonDataMap(ImmutableMap.of(
-                "info.json", infoJsonNode,
-                "foo.json", fooAnswerJsonNode,
-                "bar.json", barAnswerJsonNode,
-                "baz.json", bazAnswerJsonNode));
+        String calendarDateAnswerJsonText = "{\n" +
+                "   \"questionType\":0,\n" +
+                "   \"dateAnswer\":\"2017-01-31\",\n" +
+                "   \"startDate\":\"2015-04-02T03:23:59-07:00\",\n" +
+                "   \"questionTypeName\":\"Date\",\n" +
+                "   \"item\":\"calendar-date\",\n" +
+                "   \"endDate\":\"2015-04-02T03:24:01-07:00\"\n" +
+                "}";
+        JsonNode calendarDateAnswerJsonNode = BridgeObjectMapper.get().readTree(calendarDateAnswerJsonText);
+
+        String timeWithoutDateAnswerJsonText = "{\n" +
+                "   \"questionType\":0,\n" +
+                "   \"dateComponentsAnswer\":\"16:42:52.256\",\n" +
+                "   \"startDate\":\"2015-04-02T03:23:59-07:00\",\n" +
+                "   \"questionTypeName\":\"TimeOfDay\",\n" +
+                "   \"item\":\"time-without-date\",\n" +
+                "   \"endDate\":\"2015-04-02T03:24:01-07:00\"\n" +
+                "}";
+        JsonNode timeWithoutDateAnswerJsonNode = BridgeObjectMapper.get().readTree(timeWithoutDateAnswerJsonText);
+
+        String legacyDateTimeAnswerJsonText = "{\n" +
+                "   \"questionType\":0,\n" +
+                "   \"dateAnswer\":\"2017-01-31T16:42:52.256-0800\",\n" +
+                "   \"startDate\":\"2015-04-02T03:23:59-07:00\",\n" +
+                "   \"questionTypeName\":\"DateAndTime\",\n" +
+                "   \"item\":\"legacy-date-time\",\n" +
+                "   \"endDate\":\"2015-04-02T03:24:01-07:00\"\n" +
+                "}";
+        JsonNode legacyDateTimeAnswerJsonNode = BridgeObjectMapper.get().readTree(legacyDateTimeAnswerJsonText);
+
+        String newDateTimeAnswerJsonText = "{\n" +
```

Comment:
Good catch. My intent was the legacy type was questionTypeName=Date and the new type is questionTypeName=DateAndTime, but I guess copy-paste got the better of me. Will fix this.